diff -Naru a/autotest/meson.build b/autotest/meson.build
--- a/autotest/meson.build	2024-02-02 09:06:16.358836470 +0300
+++ b/autotest/meson.build	2024-02-02 09:07:25.828217904 +0300
@@ -2,7 +2,6 @@
                 'main.cpp')
 
 dependencies = []
-dependencies += dependency('libsystemd')
 dependencies += dependency('yaml-cpp')
 dependencies += dependency('libpcap')
 dependencies += dependency('gmock')
diff -Naru a/controlplane/main.cpp b/controlplane/main.cpp
--- a/controlplane/main.cpp	2024-02-02 09:03:36.233693019 +0300
+++ b/controlplane/main.cpp	2024-02-02 09:06:31.775363899 +0300
@@ -1,5 +1,4 @@
 #include <signal.h>
-#include <systemd/sd-daemon.h>
 
 #include <iostream>
 
@@ -58,8 +57,6 @@
 		return 3;
 	}
 
-	sd_notify(0, "READY=1");
-
 	application.start();
 	application.join();
 	return 0;
diff -Naru a/controlplane/meson.build b/controlplane/meson.build
--- a/controlplane/meson.build	2024-02-02 09:03:36.233693019 +0300
+++ b/controlplane/meson.build	2024-02-02 09:06:41.211945908 +0300
@@ -1,5 +1,4 @@
 dependencies = []
-dependencies += dependency('libsystemd')
 dependencies += dependency('threads')
 dependencies += libyanet_fwparser_dep
 dependencies += libyanet_protobuf_dep
diff -Naru a/controlplane/unittest/meson.build b/controlplane/unittest/meson.build
--- a/controlplane/unittest/meson.build	2024-02-02 09:03:36.237026318 +0300
+++ b/controlplane/unittest/meson.build	2024-02-02 09:06:51.108524241 +0300
@@ -1,5 +1,4 @@
 dependencies = []
-dependencies += dependency('libsystemd')
 dependencies += dependency('threads')
 dependencies += dependency('gtest')
 dependencies += dependency('gtest_main')
diff -Naru a/dataplane/main.cpp b/dataplane/main.cpp
--- a/dataplane/main.cpp	2024-02-02 09:03:36.237026318 +0300
+++ b/dataplane/main.cpp	2024-02-02 09:07:14.784981424 +0300
@@ -1,5 +1,4 @@
 #include <signal.h>
-#include <systemd/sd-daemon.h>
 
 #include <iostream>
 
@@ -67,8 +66,6 @@
 		return 3;
 	}
 
-	sd_notify(0, "READY=1");
-
 	dataPlane.start();
 	dataPlane.join();
 
diff -Naru a/dataplane/meson.build b/dataplane/meson.build
--- a/dataplane/meson.build	2024-02-02 09:06:16.358836470 +0300
+++ b/dataplane/meson.build	2024-02-02 09:07:45.954709410 +0300
@@ -1,6 +1,5 @@
 dependencies = []
 dependencies += dependency('libdpdk')
-dependencies += dependency('libsystemd')
 dependencies += dependency('threads')
 
 sources = files('bus.cpp',
diff -Naru a/dataplane/unittest/meson.build b/dataplane/unittest/meson.build
--- a/dataplane/unittest/meson.build	2024-02-02 09:06:16.358836470 +0300
+++ b/dataplane/unittest/meson.build	2024-02-02 09:07:03.155084001 +0300
@@ -1,6 +1,5 @@
 dependencies = []
 dependencies += dependency('libdpdk')
-dependencies += dependency('libsystemd')
 dependencies += dependency('threads')
 dependencies += dependency('gtest')
 dependencies += dependency('gtest_main')
diff -Naru a/logger/main.cpp b/logger/main.cpp
--- a/logger/main.cpp	2024-02-02 09:03:36.247026219 +0300
+++ b/logger/main.cpp	2024-02-02 09:08:09.167842817 +0300
@@ -1,6 +1,5 @@
 #include <fstream>
 #include <signal.h>
-#include <systemd/sd-daemon.h>
 #include <thread>
 
 #include <iostream>
@@ -318,7 +317,5 @@
 		return ret;
 	}
 
-	sd_notify(0, "READY=1");
-
 	return runLogger();
 }
diff -Naru a/logger/meson.build b/logger/meson.build
--- a/logger/meson.build	2024-02-02 09:06:16.358836470 +0300
+++ b/logger/meson.build	2024-02-02 09:07:53.777975116 +0300
@@ -1,6 +1,5 @@
 dependencies = []
 dependencies += dependency('libdpdk')
-dependencies += dependency('libsystemd')
 dependencies += dependency('threads')
 
 sources = files('main.cpp')
